examples/vision/text_image_denoiser.py

```diff
@@ -0,0 +1,280 @@
+the Upscaling path with skip-connections.
+"""
+
+
+def conv_unit(x, filters=64, padding="same"):
+    x = keras.layers.Conv2D(filters=filters, kernel_size=3, padding=padding)(x)
+    x = keras.layers.BatchNormalization()(x)
+    x = keras.layers.Activation("relu")(x)
+    return x
+
+
+def subpixel_upsample(x, target_channels, upscale_ratio):
+    estimated_filters = target_channels * (upscale_ratio ** 2)
+    x = keras.layers.Conv2D(estimated_filters, 3, padding="same")(x)
+    x = keras.layers.BatchNormalization()(x)
+    x = keras.layers.LeakyReLU(0.2)(x)
+    x = tf.nn.depth_to_space(x, block_size=upscale_ratio)
+    return x
+
+
+def upscale_unit(previous_input, current_input, target_channels=3, upscale_ratio=2):
+    previous_low_scale = subpixel_upsample(
+        previous_input, target_channels=target_channels, upscale_ratio=upscale_ratio
+    )
+    current_high_scale = conv_unit(current_input, filters=target_channels)
+    concat_scales = tf.concat((previous_low_scale, current_high_scale), axis=-1)
+    feature_manifold = conv_unit(concat_scales, filters=target_channels)
+    return feature_manifold
+
+
+def denoiser(height, width):
+    base_model = resnet_v2.ResNet50V2(
+        include_top=False, weights="imagenet", input_shape=(height, width, 3)
+    )
+    encode0 = base_model.get_layer("input_1").output  # (None,352,608,3)
+    encode1 = base_model.get_layer("conv1_conv").output  # (None,176,304,64)
+    encode2 = base_model.get_layer("conv2_block3_1_relu").output  # (None,88,152,64)
+    encode3 = base_model.get_layer("conv3_block4_1_relu").output  # (None,44,76,128)
+
+    decode1 = upscale_unit(
+        previous_input=encode3, current_input=encode2, target_channels=128
+    )
+    decode2 = upscale_unit(
+        previous_input=decode1, current_input=encode1, target_channels=64
+    )
+    decode3 = upscale_unit(
+        previous_input=decode2, current_input=encode0, target_channels=32
+    )
+
+    out_conv = keras.layers.Conv2D(3, 5, padding="same", activation="sigmoid")(decode3)
+    denoiser_model = keras.Model(base_model.input, out_conv, name="denoiser")
+    denoiser_model.summary()
+    return denoiser_model
+
+
+"""
+## Training
+
+The Model is Trained with an Objective of minimizing ```Mean Squared Error```.
+"""
+
+EPOCHS = 40
+LEARNING_RATE = 1e-4
+
+denoiser_net = denoiser(height=HEIGHT, width=WIDTH)
+
+early_stop = callbacks.EarlyStopping(patience=8)
+ckpt = callbacks.ModelCheckpoint(
+    "best_ckpt.h5", save_best_only=True, save_weights_only=True, verbose=1
+)
+optimizer = keras.optimizers.Adam(LEARNING_RATE)
+denoiser_net.compile(optimizer=optimizer, loss="mse", metrics=["mae"])
+
+denoiser_net.fit(
+    train_set,
+    validation_data=valid_set,
+    batch_size=BATCH_SIZE,
+    epochs=EPOCHS,
+    steps_per_epoch=len(train_source) // BATCH_SIZE,
+    validation_steps=len(val_source) // BATCH_SIZE,
+    workers=-1,
+    callbacks=[early_stop, ckpt],
+)
```

Comment:
@anish9 In the [guide-book](https://github.com/keras-team/keras-io/blob/master/contributor_guide.md#demonstrate-the-full-power-of-your-model-but-keep-the-run-time-short), it mentioned about - **Demonstrate the full power of your model, but keep the run time short**, please make sure that this example full-fill that request.

Reply:
@anish9 could you please ensure this?